### PR TITLE
[DOC] New page for strftime formats

### PR DESCRIPTION
**Pull request by BurdetteLamar**
This new page would be linked from method `strftime` in `Time`, `Date`, and `DateTime`, replacing the text there.

### CODE REVIEW

File: doc/strftime_formatting.rdoc [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    Time.new(2022, 1, 1, 13).strftime('%H:%M:%S') # => "13:00:00"
+
+- <tt>%+</tt> (not supported in Time#strftime) - \Date and time:
+
+    DateTime.now.strftime('%+')
+    # => "Wed Jun 29 08:31:53 -05:00 2022"
+    DateTime.now.strftime('%a %b %e %H:%M:%S %Z %Y')
+    # => "Wed Jun 29 08:32:18 -05:00 2022"
+
+==== Common Format Specifications
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/strftime_formatting.rdoc:298]: I could not find these in any ISO doc.

File: doc/strftime_formatting.rdoc [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+It consists of:
+
+- A leading percent character.
+- Zero or more _flags_ (each is a character).
+- An optional _width_ _specifier_ (an integer).
+- A _conversion_ _specifier_ (a character).
+
+Note that the +E+ and +O+ modifiers seen in some implementations
+are not supported here and should not be used.
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/strftime_formatting.rdoc]: Oops, removed from format and list above, but missed this.  Fixed.

File: doc/strftime_formatting.rdoc [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+==== \Date (Year, Month, Day)
+
+- <tt>%Y</tt> - Year including century, zero-padded:
+
+    Time.now.strftime('%Y')          # => "2022"
+    Time.new('-1000').strftime('%Y') # => "-1000" # Before common era.
+    Time.new('10000').strftime('%Y') # => "10000" # Far future.
+    Time.new('10').strftime('%Y')    # => "0010"  # Zero-padded by default.
+
+- <tt>%y</tt> - Year without century, range (0.99), zero-padded:
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/strftime_formatting.rdoc]: Fixed.

File: doc/strftime_formatting.rdoc [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+    DateTime.now.strftime('%+')
+    # => "Wed Jun 29 08:31:53 -05:00 2022"
+    DateTime.now.strftime('%a %b %e %H:%M:%S %Z %Y')
+    # => "Wed Jun 29 08:32:18 -05:00 2022"
+
+==== Common Format Specifications
+
+Examples in this section assume:
+
+  t = Time.now# => 2022-06-28 17:21:43.8555777 -0500
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/strftime_formatting.rdoc]: Fixed.

File: doc/strftime_formatting.rdoc [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+
+- <tt>^</tt> - Upcase result:
+
+    Time.new(2022, 1).strftime('%B')  # => "January" # No casing flag.
+    Time.new(2022, 1).strftime('%#B') # => "JANUARY"
+
+- <tt>#</tt> - Swapcase result:
+
+    Time.new(2022, 1).strftime('%B')  # => "January" # No casing flag.
+    Time.new(2022, 1).strftime('%#B') # => "JANUARY"
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/strftime_formatting.rdoc]: Fixed.

File: doc/strftime_formatting.rdoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (peterzhu2118) [reply doc/strftime_formatting.rdoc:295]: I see them here: https://en.wikipedia.org/wiki/ISO_8601

[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/strftime_formatting.rdoc:295]: I don't see them.  Search for `%Y%m%d` fails, and in fact I find only one `%` on that page.

[the remark this answers is not in the document]
Comment (peterzhu2118) [reply doc/strftime_formatting.rdoc:295]: That would be the `YYYYMMDD` format. The ISO formats don't use format specifiers.

[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/strftime_formatting.rdoc:295]: In that case, I don't think we should mention them.

[the remark this answers is not in the document]
Comment (peterzhu2118) [reply doc/strftime_formatting.rdoc:295]: These "Common Format Specifications" aren't random, they form ISO 8601 date/time strings.

File: doc/strftime_formatting.rdoc [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+    Time.now.strftime('%Y-%m-%d') # => "2022-06-29"
+
+- <tt>%v</tt> - VMS date:
+
+    Time.now.strftime('%v')         # => "29-JUN-2022"
+    Time.now.strftime('%e-%^b-%4Y') # => "29-JUN-2022"
+
+- <tt>%x</tt> - Same as <tt>%D</tt>.
+
+- <tt>%X</tt> - Same as <tt>%T</tt>.
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/strftime_formatting.rdoc:267]: `%b` and `%h` now separated.

File: doc/strftime_formatting.rdoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/strftime_formatting.rdoc]: Formats in this section now have a strong relationship to https://en.wikipedia.org/wiki/ISO_8601.

**Comment by BurdetteLamar** [comment]
@peterzhu2118, ready for review.

File: doc/strftime_formatting.rdoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (peterzhu2118) [reply doc/strftime_formatting.rdoc]: @BurdetteLamar did you miss this?

[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/strftime_formatting.rdoc]: Fixed (two places).

**Comment by BurdetteLamar** [comment]
@peterzhu2118, ready for review.

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Pull request merged**